clang/CMakeLists.txt

```diff
@@ -214,6 +214,39 @@ if(GCC_INSTALL_PREFIX AND NOT USE_DEPRECATED_GCC_INSTALL_PREFIX)
     "See https://github.com/llvm/llvm-project/pull/77537 for detail.")
 endif()
 
+if(APPLE)
+  check_include_file(xcselect.h CLANG_HAVE_XCSELECT_H)
```

Comment:
All the configuration in this block should only be done if `CLANG_USE_XCSELECT` is enabled.